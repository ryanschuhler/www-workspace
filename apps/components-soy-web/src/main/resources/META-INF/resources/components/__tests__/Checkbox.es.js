'use strict';

import Checkbox from '../input/Checkbox.es';

let component;

describe('Checkbox', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  beforeAll(() => {
		window.Liferay = {
			mdc: {
				menu: {
					MDCSimpleMenu: jest.fn()
				}
			},
			once: jest.fn()
    };
  });

  it('should render checked', () => {
    component = new Checkbox({
      id: 'test',
      checked: true
    });

    expect(component).toMatchSnapshot();
  });
});