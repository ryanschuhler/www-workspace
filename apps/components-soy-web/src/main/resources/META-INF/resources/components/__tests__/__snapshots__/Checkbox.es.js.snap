// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render checked 1`] = `
<div class="mdc-form-field">
  <div class="mdc-checkbox checked">
    <input checked="" class="mdc-checkbox__native-control" id="test" ref="input" type="checkbox">
    <div class="mdc-checkbox__background">
      <svg class="mdc-checkbox__checkmark" viewBox="0 0 24 24">
        <path class="mdc-checkbox__checkmark__path" fill="none" stroke="white" d="M1.73,12.91 8.1,19.28 22.79,4.59"></path>
      </svg>
      <div class="mdc-checkbox__mixedmark"></div>
    </div>
  </div>
</div>
`;
